src/Transcoding/Transcoder.php

```diff
@@ -39,19 +43,19 @@ public function __construct(
         $this->logger->pushHandler($fileHandler);
     }
 
+    /**
+     * @inheritdoc
+     */
     public function transcode(
-        $inputFile,
-        $outputFile,
-        $trackPreviewStart = 0,
-        $performTrim = false,
-        $clipLength = 90,
-        $fadeIn = false,
-        $fadeInLength = 6,
-        $fadeOut = false,
-        $fadeOutLength = 6
+        string $inputFile,
+        string $outputFile,
+        float $trimStartTime = 0,
+        float $clipLength = 90,
+        float $fadeInLength = 0,
+        float $fadeOutLength = 0
     ) {
```

Comment:
If opting to use `parameter types`, maybe its note worthy to mention `return types` ?

Reply:
Agreed. I've added them in a couple of places, and think it makes sense to use them consistently throughout. This is our chance to modernize the library, and there's no incentive not to take advantage of this construct.